services/storage/src/simcore_service_storage/handlers_files.py

```diff
@@ -254,6 +254,7 @@ async def complete_upload_file(request: web.Request) -> web.Response:
     query_params = parse_request_query_parameters_as(StorageQueryParamsBase, request)
     path_params = parse_request_path_parameters_as(FilePathParams, request)
     body = await parse_request_body_as(FileUploadCompletionBody, request)
+    body.sort()  # ensure uploaded parts are sorted (required by S3)
```

Comment:
If you add the validator that @sanderegg  proposes, you do not need this line

Reply:
Yes, I understand. I added it like this in case there one day is a bug and people will have to look into the code. This makes it clear that the data from the user is actually "post-processed" before "consuming" it and not just validating it.